BEST_PRACTICES.md

```diff
@@ -127,7 +127,68 @@ Multiple-lines:
  */
 ```
 
-## Blocks, delegates or data source
+## Blocks, delegates and data source
+
+## Naming
+
+Declaring delegates for some UIViewControllers, for example UITableViewControllers or UICollectionViewControllers, can be annoying, since they have their own delegates that use the variable `delegate`, and can't be overwritten.
+
+Usually this leads to the developer naming the delegate things like `recipesDelegate`, `recipesControllerDelegate` and so on. This causes to confusion for the API user, since now there are two delegates. To avoid this, we do not subclass `UITableViewController`and `UICollectionViewController` directly. Instead we use our own classes here: `SweetTableController` and `SweetCollectionController` instead. This frees up the `delegate` variable for our own use. If you are using something else than UITableViewController and UICollectionViewController, use a meaningful name for your delegate, such as the name of the protocol or the class.
+
+**Preferred:**
+
+```swift
+protocol RecipesControllerDelegate: class {
+}
+
+class RecipesController: SweetTableController/SweetCollectionController {
+    weak var delegate: RecipesControllerDelegate?
+}
+```
+
+```swift
+protocol BackToRootRecipesNavigationControllerDelegate: class {
+}
+
+class RecipesNavigationController: UINavigationController {
+    weak var backToRootDelegate: BackToRootRecipesNavigationControllerDelegate?
+}
+```
+
+```swift
+protocol NewRecipesControllerDelegate: class {
+}
+
+protocol ExistingRecipesControllerDelegate: class {
+}
+
+class RecipesController: UIPresentationViewController {
```

Comment:
This is an example of a controller that will edit or create a recipe if it's a new recipe I want a different delegate than if it's an existing (edited) recipe. That's why I have `newRecipesDelegate` and `existingRecipesDelegate`.